modules/dnn_modern/CMakeLists.txt

```diff
@@ -0,0 +1,111 @@
+
+if(TINYDNN_USE_TBB)
+    find_package(TBB REQUIRED)
+    add_definitions(-DCNN_USE_TBB)
+    include_directories(${TBB_INCLUDE_DIRS})
+    link_directories(${TBB_LIBRARY_DIRS})
+    listt(APPEND REQUIRED_LIBRARIES ${TBB_LIBRARIES})
+elseif(TINYDNN_USE_OMP)
+    find_package(OpenMP REQUIRED)
+    add_definitions(-DCNN_USE_OMP)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
+endif()
+
+if(TINYDNN_USE_NNPACK)
+   find_package(NNPACK REQUIRED)
+   add_definitions(-DCNN_USE_NNPACK)
+   include_directories(SYSTEM ${NNPACK_INCLUDE_DIR})
+   include_directories(SYSTEM ${NNPACK_INCLUDE_DIR}/../third-party/pthreadpool/include)
+   list(APPEND REQUIRED_LIBRARIES ${NNPACK_LIB})
+endif()
+
+find_package(Protobuf)
+find_package(Caffe)
+list(APPEND REQUIRED_LIBRARIES ${PROTOBUF_LIBRARIES}
+                               ${Caffe_LIBRARIES})
+
+# ----------------------------------------------------------------------------
+# COMPILER OPTIONS
+# ----------------------------------------------------------------------------
+
+if(CMAKE_COMPILER_IS_GNUCXX OR MINGW OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    if(TINYDNN_USE_SSE)
+        add_definitions(-DCNN_USE_SSE)
+        set(EXTRA_C_FLAGS  " ${EXTRA_C_FLAGS} -msse3 ")
+    endif()
+    if(TINYDNN_USE_AVX)
+        add_definitions(-DCNN_USE_AVX)
+        set(EXTRA_C_FLAGS  "${EXTRA_C_FLAGS} -mavx ")
+    endif()
+    if((NOT TINYDNN_USE_TBB) AND (NOT TINYDNN_USE_OMP))
+        set(EXTRA_C_FLAGS " ${EXTRA_C_FLAGS} -pthread ")
+    endif()
+    set(EXTRA_C_FLAGS " ${EXTRA_C_FLAGS} -std=c++11 -Wall -Wpedantic")
+    set(EXTRA_C_FLAGS_RELEASE " -O3 ")
+    set(EXTRA_C_FLAGS_DEBUG " -g3 ")
+else() # MSVC
+    if(TINYDNN_USE_SSE)
+        add_definitions(-DCNN_USE_SSE)
+        set(EXTRA_C_FLAGS  " ${EXTRA_C_FLAGS} /arch:SSE2 ")
+    endif()
+    if(TINYDNN_USE_AVX)
+        add_definitions(-DCNN_USE_AVX)
+        set(EXTRA_C_FLAGS  "${EXTRA_C_FLAGS}  /arch:AVX ")
+    endif()
+    set(EXTRA_C_FLAGS_RELEASE " /Ox /bigobj ")
+    add_definitions(-D _CRT_SECURE_NO_WARNINGS)
+endif()
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${EXTRA_C_FLAGS}")
+set(CMAKE_CXX_FLAGS_RELEASE "${EXTRA_C_FLAGS_RELEASE}")
+set(CMAKE_CXX_FLAGS_DEBUG "${EXTRA_C_FLAGS_DEBUG}")
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${EXTRA_EXE_LINKER_FLAGS}")
+set(CMAKE_EXE_LINKER_FLAGS_RELEASE "${CMAKE_EXE_LINKER_FLAGS_RELEASE} ${EXTRA_EXE_LINKER_FLAGS_RELEASE}")
+set(CMAKE_EXE_LINKER_FLAGS_DEBUG "${CMAKE_EXE_LINKER_FLAGS_DEBUG} ${EXTRA_EXE_LINKER_FLAGS_DEBUG}")
+
+# ----------------------------------------------------------------------------
+# DNN-MODERN MODULE
+# ----------------------------------------------------------------------------
+
+ocv_define_module(dnn_modern opencv_core opencv_imgproc opencv_imgcodecs WRAP python)
+ocv_target_link_libraries(${the_module} TinyDNN::tiny_dnn ${REQUIRED_LIBRARIES})
```

Comment:
I followed the instructions form README, but tiny-dnn headers are missing during compilation,
so it's needed to add implicit include of sources.
Also in case if headers are included compilation fails due to -Werror=non-virtual-dtor warning.